cscs-checks/tools/profiling_and_debugging/intel_advisor_roofline.py

```diff
@@ -0,0 +1,161 @@
+        ]
+        self.num_tasks = 1
+        self.num_tasks_per_node = 1
+        self.num_cpus_per_task = 1
+        self.variables = {
+            'OMP_NUM_THREADS': str(self.num_cpus_per_task),
+            'CRAYPE_LINK_TYPE': 'dynamic',
+        }
+        self.pre_run = [
+            # Testing with advisor/2018:
+            #   advisor/2019 is broken on dom ("Exceeded job memory limit"),
+            #   and advisor/2019 is not installed on daint,
+            'source $INTEL_PATH/../advisor_2018/advixe-vars.sh',
+            'advixe-cl -help collect |head -20',
+        ]
+        self.executable = 'advixe-cl'
+        self.exe = './roof.exe'
+        self.roofdir = './roof.dir'
+        self.executable_opts = [
+            '--collect survey --project-dir=%s --search-dir src:rp=. '
+            '--data-limit=0 --no-auto-finalize --trace-mpi -- %s ' %
+            (self.roofdir, self.exe)
+        ]
+        self.version_rpt = 'Intel_Advisor_version.rpt'
+        self.post_run = [
+            # collecting the performance data for the roofline model is a 2
+            # steps process:
+            'srun %s --collect tripcounts --flop --project-dir=%s '
+            '--search-dir src:rp=. --data-limit=0 --no-auto-finalize '
+            '--trace-mpi -- %s' % (self.executable, self.roofdir, self.exe),
+            # check tool's version:
+            'advixe-cl -V &> %s' % self.version_rpt,
+            # "advixe-cl --report" looks for e000/ in the output directory;
+            # if not found, it will fail with:
+            # IOError: Survey result cannot be loaded
+            'cd %s;ln -s nid* e000;cd -' % self.roofdir,
+        ]
+        self.roofline_ref = 'Intel_Advisor_roofline_reference.rpt'
+        self.roofline_rpt = 'Intel_Advisor_roofline_results.rpt'
+        self.post_run += [
+            # report reference values/boundaries (roofline_ref):
+            'advixe-cl --report=roofs --project-dir=%s &> %s' %
+            (self.roofdir, self.roofline_ref),
+            'python2 API/cscs.py %s &> %s' % (self.roofdir, self.roofline_rpt),
+            # 'advixe-cl --format=csv' seems to be not working (empty report),
+            # keeping as reference for a future check:
+            #   'advixe-cl --show-all-columns -csv-delimiter=";"'
+            #   ' --report=tripcounts --format=csv --project-dir=%s &> %s'
+            # This can be used instead (see advisor/config/report/roofs.tmpl):
+            #   'advixe-cl --report custom --report-template ./TEMPL/cscs.tmpl'
+            #   ' --project-dir=%s &> %s'
+        ]
+        self.maintainers = ['JG']
+        self.tags = {'production'}
+
+    def setup(self, partition, environ, **job_opts):
+        super().setup(partition, environ, **job_opts)
+        environ_name = self.current_environ.name
+        prgenv_flags = self.prgenv_flags[environ_name]
+        self.build_system.cxxflags = prgenv_flags
+        toolsversion = '551025'  # 2018 Update 2 (build 551025)
```

Comment:
This could be class attribute and defined in `__init__()`

Reply:
:ok: